src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Automation/Peers/MenuItemDataAutomationPeer.cs

```diff
@@ -0,0 +1,166 @@
+            if (wrapperPeer != null)
+            {
+                IExpandCollapseProvider expandCollapseProvider = wrapperPeer as IExpandCollapseProvider;
+                expandCollapseProvider.Expand();
+            }
+            ThrowElementNotAvailableException();
+        }
+
+        void IExpandCollapseProvider.Collapse()
+        {
+            MenuItemAutomationPeer wrapperPeer = GetWrapperPeer() as MenuItemAutomationPeer;
+            if (wrapperPeer != null)
+            {
+                IExpandCollapseProvider expandCollapseProvider = wrapperPeer as IExpandCollapseProvider;
+                expandCollapseProvider.Collapse();
+            }
+            ThrowElementNotAvailableException();
+        }
+
+        ExpandCollapseState IExpandCollapseProvider.ExpandCollapseState
+        {
+            get
+            {
+                MenuItemAutomationPeer wrapperPeer = GetWrapperPeer() as MenuItemAutomationPeer;
+                if (wrapperPeer != null)
+                {
+                    IExpandCollapseProvider expandCollapseProvider = wrapperPeer as IExpandCollapseProvider;
+                    return expandCollapseProvider.ExpandCollapseState;
+                }
+                ThrowElementNotAvailableException();
+                return ExpandCollapseState.LeafNode;
+            }
+        }
+
+        #endregion
+
+        #region IInvokeProvider Members
+
+        void IInvokeProvider.Invoke()
+        {
+            // check if enabled
+            EnsureEnabled();
+
+            MenuItemAutomationPeer wrapperPeer = GetWrapperPeer() as MenuItemAutomationPeer;
+            if (wrapperPeer != null)
+            {
+                IInvokeProvider invokeProvider = wrapperPeer as IInvokeProvider;
+                invokeProvider.Invoke();
+            }
+            else
+            {
+                ThrowElementNotAvailableException();
+            }
+        }
+
+        #endregion
+
+        #region IToggleProvider Members
+
+        void IToggleProvider.Toggle()
+        {
+            MenuItemAutomationPeer wrapperPeer = GetWrapperPeer() as MenuItemAutomationPeer;
+            if (wrapperPeer != null)
+            {
+                IToggleProvider toggleProvider = wrapperPeer as IToggleProvider;
+                toggleProvider.Toggle();
+            }
+            else
+            {
+                ThrowElementNotAvailableException();
+            }
+        }
+
+        ToggleState IToggleProvider.ToggleState
+        {
+            get
+            {
+                MenuItemAutomationPeer wrapperPeer = GetWrapperPeer() as MenuItemAutomationPeer;
+                if (wrapperPeer != null)
+                {
+                    IToggleProvider toggleProvider = wrapperPeer as IToggleProvider;
+                    return toggleProvider.ToggleState;
+                }
+                ThrowElementNotAvailableException();
+                return ToggleState.Indeterminate;
+            }
+        }
```

Comment:
```suggestion
        ToggleState IToggleProvider.ToggleState
        {
            get
            {
                MenuItemAutomationPeer wrapperPeer = GetWrapperPeer() as MenuItemAutomationPeer;
                if (wrapperPeer is IToggleProvider toggleProvider)
                {
                    return toggleProvider.ToggleState;
                }
                ThrowElementNotAvailableException();
                return ToggleState.Indeterminate;
            }
        }
```

And I do not think throw the exception is a good idea here.

Reply:
This code was adapted from the implementation of `IExpandCollapseProvider.ExpandCollapseState` in `TreeViewDataItemAutomationPeer`. I am actually not sure if it is even possible that the wrapped peer can be something other than `MenuItemAutomationPeer` during runtime. May when the type is used by a custom implementation in a wrong way.